File: backend/Whale.DAL/Models/GroupMessage.cs
using System;
using System.ComponentModel.DataAnnotations;
using Whale.DAL.Abstraction;

namespace Whale.DAL.Models
{
    public class GroupMessage : BaseEntity
    {
        public Guid GroupId { get; set; }
        public Group Group { get; set; }
        public Guid AuthorId { get; set; }
        public User Author { get; set; }
        public DateTimeOffset CreatedAt { get; set; }
        [Required]
        public string Message { get; set; }
        public bool Attachment { get; set; }
        public string AttachmentUrl { get; set; }
    }
}

File: frontend/src/app/shared/models/message/group-message.ts
import { Group } from '../group/group';
import { User } from '@shared/models/user/user';

export interface GroupMessage {
  id?: string;
  groupId: string;
  group?: Group;
  authorId: string;
  author?: User;
  createdAt: Date;
  message: string;
  attachment: boolean;
  attachmentUrl: string;
}
